### PR TITLE
AP_HAL_ChibiOS:modify hwdef for V3 of ReaperF745

### PR DESCRIPTION
**Pull request by Hwurzburg**
now either rename to just ReaperF745  or split changes into a new hwdef based off the v2...I prefer the former
tested on V3 board....has the same funky 45 deg vehicle forward as the V2...

I sent Andy P a link to the build to test on his V2 to make sure it still works

### CODE REVIEW

File: libraries/AP_HAL_ChibiOS/hwdef/ReaperF745v2/hwdef.dat [hunk 1]
```diff
@@ -118,17 +118,20 @@ STORAGE_FLASH_PAGE 1
 # enable logging to dataflash
 define HAL_LOGGING_DATAFLASH_ENABLED 1
 
-# spi devices
-SPIDEV bmi270    SPI3 DEVID1 BMI270_CS  MODE3  1*MHZ  4*MHZ
+# spi devices bmi270(v2) icm42688 (v3)
+SPIDEV bmi270    SPI3 DEVID1 GYRO_CS  MODE3  1*MHZ  4*MHZ
+SPIDEV icm42688  SPI3 DEVID1 GYRO_CS  MODE3  2*MHZ  8*MHZ
```
Comment (andyp1per) [inline libraries/AP_HAL_ChibiOS/hwdef/ReaperF745v2/hwdef.dat:123]: On all the other ports of this nature I have done it needs to be DEVID2
Also you should be able to go up to 16Mhz for the 42688

Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/ReaperF745v2/hwdef.dat:123]: its not necessary if they cant exist at the same time AIFAIK  ... the MAtekH743 does not require it

Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/ReaperF745v2/hwdef.dat:123]: every other instantiation is 8Mhz, except the MatekH743...dont know if H7 makes a difference or not

**Comment by Hwurzburg** [comment]
renamed rather than create a separate V3 hwdef

**Comment by Hwurzburg** [comment]
space was down to 2K, free space by minimizing GPS and Airspeed backends

File: libraries/AP_HAL_ChibiOS/hwdef/ReaperF745v2/hwdef.dat (hunk 1 above)
Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/ReaperF745v2/hwdef.dat:123]: > its not necessary if they cant exist at the same time AIFAIK ... the MAtekH743 does not require it

Just saying what was required on other boards ...

Comment (andyp1per) [reply libraries/AP_HAL_ChibiOS/hwdef/ReaperF745v2/hwdef.dat:123]: > every other instantiation is 8Mhz including the ones you just did, except the MatekH743...dont know if H7 makes a difference or not

Nora is 16Mhz. Not sure why you are pushing back against this Henry? Faster is better if the board supports it. Depending on the clock the actual selected speed will be somewhat lower. Please try 16Mhz.

**Review by andyp1per — CHANGES_REQUESTED** [review]
Please try 16Mhz

File: libraries/AP_HAL_ChibiOS/hwdef/ReaperF745v2/hwdef.dat (hunk 1 above)
Comment (Hwurzburg) [reply libraries/AP_HAL_ChibiOS/hwdef/ReaperF745v2/hwdef.dat:123]: I am just going by previous patterns....not knowing anything else....including when you did the MambaF405 with this IMU..you set it at 8MHz not 16MHz..the only hwdefs which set it at 16MHZ were H7 processors...no F4 boards have it set....just previous patterns, not knowledge....happy to change to 16MHZ if you are sure its better....but then we should update all the previous boards using it at 8MHZ, correct?

Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/ReaperF745v2/hwdef.dat:123]: DEVID1 is fine, as on the same CS pin, so can't have both at once

**Comment by Hwurzburg** [comment]
> Please try 16Mhz

@andyp1per we discussed  in devcall and it was felt that trying this on a single board does not prove the board layout over production is guaranteed to support it, hence the use of a more conservative clock on most boards...those boards that do use it had OEMs that verified on multiple boards....and there was the uncertainty of real benefit...

have you had a chance to test on a v2 board?

**Comment by andyp1per** [comment]
> we discussed in devcall and it was felt that trying this on a single board does not prove the board layout over production is guaranteed to support it, hence the use of a more conservative clock on most boards...those boards that do use it had OEMs that verified on multiple boards....and there was the uncertainty of real benefi

The issue is that because of the clock tree the actual speed on the boards I have tested come out significantly lower that 8Mhz. Setting to 16Mhz usually ends up with a speed of about 12Mhz, which is conservative enough but much better than the 6Mhz or so that you get otherwise. This is a small board with short traces - as far as I am concerned picking 8Mhz is just as much finger in the air as anything else. However, the reduction in CPU duty cycle when you go very higher clock speeds is very significant.

I've not had a chance to try with v2 as yet.

**Comment by tridge** [comment]
> However, the reduction in CPU duty cycle when you go with higher clock speeds is very significant.

there is no change in CPU load. When waiting for the SPI completion interrupt the thread is suspended, so no CPU usage. Note that if you use the disable-interrupts method to try to measure CPU then you will see higher CPU, but that is because the CPU is artificially locked to the thread.

**Review by andyp1per — APPROVED** [review]

**Pull request merged**